octobot_trading/exchanges/connectors/cryptofeed_websocket_connector.py

```diff
@@ -117,15 +114,45 @@ def subscribe_feeds(self):
             if self.callback_by_feed.get(channel)
         }
         self.client.add_feed(self.get_feed_name(),
-                             symbols=exchange_symbols,
+                             symbols=self.pairs,
                              channels=self.channels,
                              callbacks=callbacks)
+        for channel in self.channels:
+            self.logger.debug(f"Subscribed to {channel}")
+
+    def _ensure_feed_parameters(self):
+        exchange = cryptofeed_exchanges.EXCHANGE_MAP[self.get_feed_name()](
+            config=self.client.config
+        )
+        self._ensure_symbols(exchange)
+        self._ensure_time_frames(exchange)
+
+    def _ensure_symbols(self, exchange):
+        filtered_pairs = []
+        for pair in self.pairs:
+            if pair in exchange.normalized_symbol_mapping:
+                filtered_pairs.append(pair)
+            else:
+                formatted_pair = pair.replace(self.CRYPTOFEED_DEFAULT_MARKET_SEPARATOR, octobot_commons.MARKET_SEPARATOR)
+                self.logger.error(f"{formatted_pair} pair is not supported by this exchange's websocket")
+        self.pairs = filtered_pairs
+
+    def _ensure_time_frames(self, exchange):
+        if hasattr(exchange, "valid_candle_intervals"):
+            filtered_time_frames = []
+            for time_frame in self.time_frames:
+                if time_frame.value in exchange.valid_candle_intervals:
+                    filtered_time_frames.append(time_frame)
+                else:
+                    self.logger.error(f"{time_frame.value} time frame is not supported by this exchange's websocket")
+            self.time_frames = filtered_time_frames
 
     def start(self):
         try:
+            self._ensure_feed_parameters()
             self.subscribe_feeds()
         except Exception as e:
-            self.logger.error(f"Failed to subscribe when creating websocket feed : {e}")
+            self.logger.exception(e, True, f"Failed to subscribe when creating websocket feed : {e}")
```

Comment:
Why ?

Reply:
To have the traceback when something goes wrong as it's manly here that something can happen and it's easier to debug later (and changes nothing for the user)